app/controllers/tickets_controller.rb

```diff
@@ -37,17 +39,29 @@ def destroy
     end
   end
 
-  def state # change ticket state
-      if @ticket.state_can_be_change_by?(current_user)
-        if @ticket.update(state: params[:state])
-          render json: TicketBlueprint.render(@ticket)
-        else
-          render json: {errors: @ticket.errors.full_messages}
-        end
-      else
-        render json: {message: "you have not access!"}, status: 401
-      end
+  def dev_state # change ticket state for developers
+    state(@ticket.move_up!, @ticket.can_move_up?) if @ticket.can_move_up?
+    notification("pending")
+    @ticket.notify_manager if @ticket.state == "waiting_for_accept"
+  end
+
+  def to_progress
+    state(@ticket.to_progress!, @ticket.can_to_progress?) if @ticket.can_to_progress?
```

Comment:
Redundant argument
`state(@ticket.to_progress!, @ticket.can_to_progress?) if @ticket.can_to_progress?` will be called only in case if ` @ticket.can_to_progress?` is `true` so second argument is also `@ticket.can_to_progress?` always true.